src/main/java/com/palominolabs/metrics/guice/listener/CountedListener.java

```diff
@@ -13,16 +17,18 @@
 public class CountedListener extends DeclaredMethodsTypeListener {
     private final MetricRegistry metricRegistry;
     private final MetricNamer metricNamer;
+    private final AnnotationProvider annotationProvider;
 
-    public CountedListener(MetricRegistry metricRegistry, MetricNamer metricNamer) {
+    public CountedListener(MetricRegistry metricRegistry, MetricNamer metricNamer, AnnotationProvider annotationProvider) {
         this.metricRegistry = metricRegistry;
         this.metricNamer = metricNamer;
+        this.annotationProvider = annotationProvider;
     }
 
     @Nullable
     @Override
     protected MethodInterceptor getInterceptor(Method method) {
-        final Counted annotation = method.getAnnotation(Counted.class);
+        final Counted annotation = annotationProvider.getAnnotation(method, Counted.class);
         if (annotation != null) {
             final Counter counter = metricRegistry.counter(metricNamer.getNameForCounted(method, annotation));
```

Comment:
With the more generic approach I described, it may be necessary to have a different abstraction for handing off metric configuration to the namer than just the annotation since you could have an implementation that didn't require the use of the stock metrics annotations (`@Counted` etc) at all.